### PR TITLE
Fix locking for stealth addressing

### PR DESCRIPTION
**Pull request by marpme**

### CODE REVIEW

File: src/qt/walletmodel.cpp [hunk 1]
```diff
@@ -387,6 +387,7 @@ bool WalletModel::setWalletLocked(bool locked, const SecureString &passPhrase)
     if(locked)
     {
         // Lock
+        m_wallet->lockStealthAddresses();
```
Comment (marpme) [inline src/qt/walletmodel.cpp]: locks stealth addresses after having a pop-up opened

File: src/wallet/rpcwallet.cpp [hunk 2]
```diff
@@ -2632,6 +2632,7 @@ static void LockWallet(CWallet* pWallet)
     LOCK(pWallet->cs_wallet);
     pWallet->nRelockTime = 0;
     pWallet->Lock();
+    pWallet->LockStealthAddresses();
```
Comment (marpme) [inline src/wallet/rpcwallet.cpp]: This is for locking stealth address when the timer runs out (after unlocking for a time amount)

File: src/wallet/rpcwallet.cpp [hunk 3]
```diff
@@ -2794,6 +2795,7 @@ static UniValue walletlock(const JSONRPCRequest& request)
     }
 
     pwallet->Lock();
+    pwallet->LockStealthAddresses();
```
Comment (marpme) [inline src/wallet/rpcwallet.cpp]: this takes care of locking the stealth when locking it via console

**Comment by justinvforvendetta** [comment]
this was easy to review.. how did you catch this? great job!

**Pull request closed without merging**